miniscope_io/stream_daq.py

```diff
@@ -112,48 +125,58 @@ def nbuffer_per_fm(self) -> int:
             self._nbuffer_per_fm = len(self.buffer_npix)
         return self._nbuffer_per_fm
 
-    def _parse_header(
-        self, buffer: Bits, truncate: Literal["preamble", "header", False] = False
-    ) -> Tuple[BufferHeader, bytes]:
+    def _parse_header(self, buffer: bytes) -> Tuple[BufferHeader, np.ndarray]:
         """
         Function to parse header from each buffer.
 
         Parameters
         ----------
         buffer : bytes
             Input buffer.
-        truncate : Literal[preamble, header, False], optional
-            Whether the parsed header should be truncated from the returned buffer.
-            If `"preamble"`, then only the preamble is truncated.
-            If `"header"`, then the full header is truncated.
-            If `False`, then `buffer` is returned untouched.
 
         Returns
         -------
-        Tuple[BufferHeader, bytes]
-            The returned header data and (optionally truncated) buffer data.
+        Tuple[BufferHeader, ndarray]
+            The returned header data and payload (uint8).
         """
-        pre = Bits(self.preamble)
-        if self.config.LSB:
-            pre = pre[::-1]
-        pre_len = len(pre)
-        assert buffer[:pre_len] == pre
+
+        header, payload = BufferFormatter.bytebuffer_to_ndarrays(
+            buffer=buffer,
+            header_length_words=int(self.config.header_len / 32),
+            preamble_length_words=int(len(Bits(self.config.preamble)) / 32),
+            reverse_header_bits=True,
+            reverse_header_bytes=True,
+            reverse_body_bits=True,
+            reverse_body_bytes=True,
```

Comment:
Partially yes. Ideally, I'd want all four of these exposed when developing a hardware module, but that seemed like too much to put into the YAML file, so I stopped thinking there. These should be with the buffer setting stuff in the YAML config file and this config is too awkward to be a default setting anyway, so it might actually make sense to expose these.

Reply:
ok yes - let's think about what should go in the config file - like presumably these depend on the firmware/bitfile on the FPGA, so we should think about a *device* config that is 1:1 what is needed to run a given device&firmware combo, and a *user* config that sets stuff like capture parameters etc. that can vary with the same device. what would make sense to handle in this PR? which of these should be dependent on the `LSB` setting, if any?

Reply:
That YAML configuration is technically a device configuration now, and I think we can assume user inputs will go through CLI options, at least for now. So, I guess I'll throw all four of these flags into the config.
I think users should eventually load this config with a device alias (so it's more like `-d miniscope-vx`), but we can also do this after we have a beta device with a relatively settled device config.

> which of these should be dependent on the LSB setting, if any?

I have no idea because I just removed those first and hardcoded this substitute. I'll just remove this LSB option.

Reply:
haha ok makes sense, up to you :). trying to keep PRs small-ish so we don't need to solve everything every time - ideally we don't invent a bunch of tech debt for ourselves along the way, but since I don't have a good sense of how general this needs to be yet, throwing them in the config with some documentation for now works for me. 

We'll need to do some refactoring of the configs before we do anything like a 1.0, because soon we'll be bringing in the v4 and etc. configs as well - so we don't need to be tied too closely to those, but we should probably try and match them and write migration code where needed (eg. for people who have their own custom config file, if we change a value name/interpretation provide some script for updating theirs to a new version).

point being i'm fine with it for now, we'll need to change it even more later :)